process/factory/shard/interceptorsContainerFactory.go

```diff
@@ -105,6 +111,7 @@ func NewInterceptorsContainerFactory(
 		Signer:           singleSigner,
 		AddrConv:         addrConverter,
 		FeeHandler:       txFeeHandler,
+		Trie:             stateTrie,
```

Comment:
Why do we need the state trie here? Also, if really needed, couldn't have been just a trie interface subset?

Reply:
Changed with db.